main.tf

```
@@ -117,7 +117,7 @@ resource "aws_elasticache_replication_group" "default" {
 
   auth_token                 = var.transit_encryption_enabled ? var.auth_token : null
   replication_group_id       = var.replication_group_id == "" ? module.this.id : var.replication_group_id
-  description                = module.this.id
+  description                = var.description == "" ? module.this.id : var.description
```

```suggestion
  description                = coalesce(var.description, module.this.id)
```
@pen-pal see if this helps